app/models/apple/episode.rb

```diff
@@ -2,14 +2,59 @@
 
 module Apple
   class Episode
-    attr_accessor :show, :feeder_episode, :api
+    include Apple::ApiWaiting
+    include Apple::ApiResponse
+    attr_accessor :show, :feeder_episode, :api, :api_response
+
+    AUDIO_ASSET_FAILURE = "FAILURE"
+    AUDIO_ASSET_SUCCESS = "SUCCESS"
+
+    def self.wait_for_asset_state(api, eps)
+      wait_for(eps) do |remaining_eps|
+        Rails.logger.info("Probing for episode audio asset state")
+        unwrapped = get_episodes(api, remaining_eps)
+
+        remote_ep_by_id = unwrapped.map { |row| [row["request_metadata"]["guid"], row] }.to_h
+        remaining_eps.each { |ep| ep.api_response = remote_ep_by_id[ep.guid] }
```

Comment:
This is the new polling procedure on the episode.

It was a bit problematic to set this up given how we were fetching apple episode state. Originally we were querying the _show_ to get the episode json.  This PR moves the episode state from the Apple::Show model into the Apple::Episode. But in order to do that I had to "disconnect" the apple episode json from the show's accessor methods.

Now the Apple::Episode has an `api_response` attribute (exactly the same as PodcastContainer, PodcastDelivery, and PodcastDeliveryFile) -- though it is not persisted to the database as the Apple::Episode is not an active record model.